compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -617,9 +618,22 @@ public void visit(BLangStruct structNode) {
     }
 
     public void visit(BLangObject objectNode) {
+        objectNode.fields.stream()
```

Comment:
Can we also analyze and report situations where a private object contains public fields?